src/types.ts

```diff
@@ -1,50 +1,48 @@
-// `null` is used for a hack to manage shared state with SWR
-// https://github.com/vercel/swr/pull/918
-export type fetcherFn<Data> = ((...args: any) => Data | Promise<Data>) | null
+export type fetcherFn<Data> = (...args: any) => Data | Promise<Data>
 export interface ConfigInterface<
   Data = any,
   Error = any,
   Fn extends fetcherFn<Data> = fetcherFn<Data>
 > {
-  errorRetryInterval?: number
+  errorRetryInterval: number
```

Comment:
These values are having default values as `defaultConfig`, so we don't have to make the types optional.